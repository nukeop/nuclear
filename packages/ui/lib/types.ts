export type Track = {
  local?: boolean;
  album?: string;
  artist?: { name: string } | string;
  duration?: number | string;
  position?: number | string;
  playcount?: number | string;
  title?: string;
  name?: string;
  thumbnail?: string;
};
