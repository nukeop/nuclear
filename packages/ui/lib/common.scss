$bgdarker: #1F212D;
$bgdark: #2C2E3A;
$bgdefault: #424557;
$bglight: #555977;
$bglighter: #6A6F95;

$background: $bgdark;
$background2: $bgdefault;
$background3: $bgdarker;
$lightbg: $bglighter;

$grey: #f8f8f2;
$blue: #6272a4;
$cyan: #8be9fd;
$green: #3CDD65;
$orange: #ffb86c;
$pink: #FC83D3;
$purple: #bd93f9;
$red: #ff5555;
$yellow: #f1fa8c;
$white: #fafafa;
$black: #202020;

$very-short-duration: 0.1s;
$short-duration: 0.3s;
$medium-duration: 0.5s;
$long-duration: 1s;

$mid_screen: 800px;
$small_screen: 600px;

@mixin roundCorners {
  border-radius: 0.25em;
}

@mixin circle {
  border-radius: 9999px;
  overflow: hidden;
}

@mixin transition {
  transition: 0.25s;
}

@mixin center {
  align-items: center;
  justify-content: center;
}

@mixin ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@mixin shadow {
  box-shadow: 0 0 38px rgba(0, 0, 0, 0.3), 0 0 12px rgba(0, 0, 0, 0.22) !important;
}

.nuclear {
  font-family: "Lato", sans-serif;
  color: $white;
}

@keyframes fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes fade-out {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}
