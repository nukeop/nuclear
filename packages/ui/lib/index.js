import '../resources/Lato.css';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Card } from './components/Card';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as SmoothImage } from './components/SmoothImage';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as PopupButton } from './components/PopupButton';
export { default as QueueItem } from './components/QueueItem';
export { default as QueuePopup } from './components/QueuePopup';
export { default as Range } from './components/Range';
export { default as Seekbar } from './components/Seekbar';
export { default as ToastContainer } from './components/ToastContainer';
export { default as TrackRow } from './components/TrackRow';
export { default as UserPluginsItem } from './components/UserPluginsItem';

export * from './utils';
