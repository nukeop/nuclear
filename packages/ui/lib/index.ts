import '../resources/Lato.scss';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as ArtistGrid } from './components/ArtistGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Button } from './components/Button';
export { default as Card } from './components/Card';
export { default as Checkbox } from './components/Checkbox';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as Dropdown } from './components/Dropdown';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as MiniPlayer } from './components/MiniPlayer';
export { default as PlayerBar } from './components/PlayerBar';
export { default as PlayerButton } from './components/PlayerButton';
export { default as PopupButton } from './components/PopupButton';
export { default as PopupDropdown } from './components/PopupDropdown';
export { default as QueueItem } from './components/QueueItem';
export { default as Range } from './components/Range';
export { default as SearchBox } from './components/SearchBox';
export { default as Seekbar } from './components/Seekbar';
export { default as SmoothImage } from './components/SmoothImage';
export { default as StreamInfo } from './components/StreamInfo';
export { default as ToastContainer } from './components/ToastContainer';
export { default as Tooltip } from './components/Tooltip';
export { default as TrackPopup } from './components/TrackPopup';
export { default as TrackRow } from './components/TrackRow';
export { default as TrackTable } from './components/TrackTable';
export { default as UserPluginsItem } from './components/UserPluginsItem';
export { default as Visualizer } from './components/Visualizer';
export { default as VisualizerOverlay } from './components/VisualizerOverlay';

export * from './utils';
