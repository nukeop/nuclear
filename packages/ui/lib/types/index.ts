export type Track = {
  uuid?: string;
  loading?: boolean;
  error?: boolean | { 
    message: string; 
    details: string 
  };
  local?: boolean;
  artist: { name: string } | string;
  name?: string;
  title?: string;
  album?: string;
  duration?: number | string;
  position?: number | string;
  playcount?: number | string;
  thumbnail?: string;
  image?: { '#text'?: string }[];
  streams?: TrackStream[];
};

export type Album = {
  artist?: { name: string };
  title: string;
  tracks?: Track[];
  image?: { '#text'?: string }[];
};

export type TrackStream = {
  source: string;
  id: string;
  duration?: number;
  title?: string;
  thumbnail?: string;
  stream?: string;
};

export type SearchProviderOption = {
  key: string;
  text: string;
  value: string;
}

export type TrackItem = {
  artist: string;
  name: string;
  thumbnail?: string;
  local?: boolean;
  uuid?: string;
  streams?: TrackStream[];
}

export type Download = {
  status: DownloadStatus;
  completion: number;
  track: TrackItem;
}
export enum DownloadStatus {
  WAITING = 'Waiting',
  STARTED = 'Started',
  PAUSED = 'Paused',
  FINISHED = 'Finished',
  ERROR = 'Error'
}

export type Notification = {
  id?:string
  title?: string;
  details?: string;
  error?: boolean;
  warning?: boolean;
  info?: boolean;
  success?: boolean;
  onClick?:() => void
  icon?:  Node | React.ReactElement<{
    src: string;
  }>
}
