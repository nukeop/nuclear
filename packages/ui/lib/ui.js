'use strict';

module.exports = ui;

function ui() {
  // TODO
}
