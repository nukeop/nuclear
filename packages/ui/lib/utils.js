import _ from 'lodash';

export function formatDuration(duration) {
  if (!_.isFinite(parseInt(duration)) || duration < 0) {
    return '00:00';
  }

  let sec_num = parseInt(duration, 10);
  let hours   = Math.floor(sec_num / 3600);
  let minutes = Math.floor((sec_num - (hours * 3600)) / 60);
  let seconds = sec_num - (hours * 3600) - (minutes * 60);

  if (hours < 10) {
    hours = '0' + hours;
  }
  if (minutes < 10) {
    minutes = '0' + minutes;
  }
  if (seconds < 10) {
    seconds = '0' + seconds;
  }

  if (hours === '00') {
    return minutes+':'+seconds;
  } else {
    return hours+':'+minutes+':'+seconds;
  }
}

export const getThumbnail = album => {
  return _.get(album, 'images[0].uri',
    _.get(album, 'image[0][\'#text\']',
      _.get(album, 'thumb')));
};

export const getTrackItem = track => ({
  artist: track.artist.name,
  name: track.name,
  thumbnail: getThumbnail(track),
  local: track.local,
  streams: track.streams
});
