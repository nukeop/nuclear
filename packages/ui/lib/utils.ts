import _ from 'lodash';
import { Track } from './types';

export function formatDuration(duration) {
  if (!_.isFinite(parseInt(duration)) || duration <= 0) {
    return '00:00';
  }

  const secNum = parseInt(duration, 10);
  let hours: number | string = Math.floor(secNum / 3600);
  let minutes: number | string = Math.floor((secNum - (hours * 3600)) / 60);
  let seconds: number | string = secNum - (hours * 3600) - (minutes * 60);

  if (hours < 10) {
    hours = '0' + hours;
  }
  if (minutes < 10) {
    minutes = '0' + minutes;
  }
  if (seconds < 10) {
    seconds = '0' + seconds;
  }

  if (hours === '00') {
    return minutes + ':' + seconds;
  } else {
    return hours + ':' + minutes + ':' + seconds;
  }
}

export const getTrackTitle = (track: Track) => track?.name || track?.title;

export const getTrackArtist = (track: Track) => _.isString(track?.artist)
  ? track?.artist
  : track?.artist?.name;

export const getThumbnail = albumOrTrack => _.get(albumOrTrack, 'coverImage')
  || _.get(albumOrTrack, 'thumb')
  || _.get(albumOrTrack, 'thumbnail');

export const getTrackItem = track => ({
  artist: getTrackArtist(track),
  name: getTrackTitle(track),
  thumbnail: getThumbnail(track),
  local: track.local,
  streams: track.streams,
  uuid: track.uuid
});

export const areTracksEqualByName = (trackA: Track, trackB: Track) => getTrackArtist(trackA) === getTrackArtist(trackB) && getTrackTitle(trackA) === getTrackTitle(trackB);
