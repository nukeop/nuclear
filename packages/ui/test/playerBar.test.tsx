import { SemanticICONS } from 'semantic-ui-react/dist/commonjs/generic';

import { makeSnapshotTest } from './helpers';
import { PlayerBar } from '../lib';

makeSnapshotTest(
  PlayerBar, 
  { 
    renderTrackDuration: true,
    timePlayed: '-3:14',
    timeToEnd: '2:43',
    fill: 66,
    track: 'Test song',
    artist: 'Test artist',
    cover: 'https://i.imgur.com/4euOws2.jpg',
    volume: 60,
    queue: { queueItems: [] },
    playOptions: [
      { icon: ('repeat' as SemanticICONS), enabled: false, name: 'Repeat' },
      { icon: ('magic' as SemanticICONS), name: 'Autoradio' },
      { icon: ('random' as SemanticICONS), enabled: false, name: 'Shuffle' }
    ],
    isMuted: false
  }, 
  '(Snapshot) Player bar'
);