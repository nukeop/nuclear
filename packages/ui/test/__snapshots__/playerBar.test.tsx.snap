// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Player bar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="nuclear player_bar"
  >
    <div
      class="nuclear seekbar"
    >
      <div
        class="nuclear seekbar_progress"
        style="width: 66%;"
      />
    </div>
    <div
      class="player_bar_bottom"
    >
      <div
        class="track_info"
      >
        <div
          class="nuclear cover_container"
        >
          <img
            src="https://i.imgur.com/4euOws2.jpg"
          />
        </div>
      </div>
      <div
        class="player_controls"
      >
        <button
          class="player_button"
        >
          <i
            aria-hidden="true"
            class="step backward large inverted icon"
          />
        </button>
        <button
          class="player_button"
        >
          <i
            aria-hidden="true"
            class="play big inverted icon"
          />
        </button>
        <button
          class="player_button"
        >
          <i
            aria-hidden="true"
            class="step forward large inverted icon"
          />
        </button>
      </div>
      <div
        class="volume_controls_container"
      >
        <div
          class="play_options"
        >
          <i
            aria-hidden="true"
            class="repeat large icon play_option_icon disabled"
          />
          <i
            aria-hidden="true"
            class="magic large icon play_option_icon"
          />
          <i
            aria-hidden="true"
            class="random large icon play_option_icon disabled"
          />
        </div>
        <div
          class="volume_controls"
        >
          <div
            class="volume_icon"
          >
            <i
              aria-hidden="true"
              class="volume up large icon"
            />
          </div>
          <div
            class="volume_slider"
          >
            <div
              class="range"
              style="width: 100%;"
            >
              <div
                class="baseDiv"
                style="height: 12px;"
              >
                <div
                  class="track"
                  style="border-radius: 4px; background: rgb(68, 71, 90); top: 4px; height: 4px;"
                />
                <div
                  class="fill"
                  style="border-radius: 4px; background: rgb(248, 248, 242); width: calc(100% * 0.6 + 4.800000000000001px); top: 4px; height: 4px;"
                />
                <div
                  class="thumb"
                  style="width: 12px; height: 12px; border-radius: 12px; background: rgb(248, 248, 242);"
                />
                <input
                  class="input"
                  max="100"
                  min="0"
                  style="top: 4px; height: 12px; width: calc(100% - 12px); margin-left: 6px; margin-right: 6px;"
                  type="range"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
